### PR TITLE
Add clone script

### PR DESCRIPTION
**Pull request by lanza**

### CODE REVIEW

File: scripts/clone.bash [hunk 1]
```diff
@@ -0,0 +1,87 @@
...
+    lld \
+    )
+
+  local unused_runtime_repositories=( \
+    compiler-rt \
+    libunwind \
+    libcxxabi \
+    libcxx \
+    openmp \
+    )
```
Comment (compnerd) [inline scripts/clone.bash]: The trailing `\` are not needed, please remove them.

File: scripts/clone.bash [hunk 2]
```diff
@@ -0,0 +1,87 @@
+#!/usr/bin/env bash
+
+function main() {
+  local multiple_master_repositories=( \
+    llvm \
+    clang \
+    lldb \
+    )
```
Comment (compnerd) [inline scripts/clone.bash]: Please unindent the closing `)`.

File: scripts/clone.bash [hunk 3]
```diff
@@ -0,0 +1,87 @@
...
+  declare -A pids
+
+  function do_multiple() {
+    git clone https://git.llvm.org/git/${1}.git ${2}
+    git -C ${2} remote rename origin upstream
+    git -C ${2} remote add swift https://github.com/apple/swift-${1}
+  }
+
+  for repo in ${multiple_master_repositories[@]}; do
+    local CLONE_DIR=${toolchain_source}/${repo}
```
Comment (compnerd) [inline scripts/clone.bash]: s/CLONE_DIR/GIT_DIR/?

File: scripts/clone.bash [hunk 4]
```diff
@@ -0,0 +1,87 @@
...
+    if test -d ${CLONE_DIR}; then
+      echo "${repo} already exists"
+    else
+      do_llvm ${repo} ${CLONE_DIR} &
+      pids[${repo}]=$!
+    fi
+  done
+
+  for repo in "${!pids[@]}"; do
+    wait ${pids[${repo}]}
```
Comment (compnerd) [inline scripts/clone.bash]: Doesn't this become pretty heavy given that it will more or less pull all the repositories at once.

**Review by compnerd — CHANGES_REQUESTED** [review]

File: scripts/clone.bash (hunk 3 above)
Comment (lanza) [reply scripts/clone.bash]: -Wpedantic-compnerd

File: scripts/clone.bash (hunk 4 above)
Comment (lanza) [reply scripts/clone.bash]: Ran it on my P720 and my MBP and it worked fine.

**Pull request closed without merging**